keps/168-2-pending-workloads-visibility/README.md

```diff
@@ -0,0 +1,328 @@
+    - [Story 2](#story-2)
+    - [Story 3](#story-3)
+  - [Risks and Mitigations](#risks-and-mitigations)
+    - [DDoS](#ddos)
+    - [Payload size](#payload-size)
+- [Design Details](#design-details)
+  - [API Details](#api-details)
+  - [API endpoints:](#api-endpoints)
+    - [List pending workloads in ClusterQueue](#list-pending-workloads-in-clusterqueue)
+    - [List pending workloads in LocalQueue](#list-pending-workloads-in-localqueue)
+    - [Fetch information about a single pending Workload](#fetch-information-about-a-single-pending-workload)
+  - [API Objects:](#api-objects)
+  - [Future extensions](#future-extensions)
+  - [Test Plan](#test-plan)
+    - [Overview](#overview)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+    - [E2E tests](#e2e-tests)
+  - [Graduation Criteria](#graduation-criteria)
+    - [GA](#ga)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [Alternative approaches](#alternative-approaches)
+    - [Approach described in <a href="https://github.com/kubernetes-sigs/kueue/tree/main/keps/168-pending-workloads-visibility">KEP#168</a>](#approach-described-in-kep168)
+    - [Extend API using CRDs](#extend-api-using-crds)
+  - [Alternatives within the proposal](#alternatives-within-the-proposal)
+    - [apiserver-builder library](#apiserver-builder-library)
+<!-- /toc -->
+
+## Summary
+
+This KEP proposes to introduce a new API that allows users to on-demand fetch information about pending workloads in both ClusterQueue and LocalQueue. Users will be able to look up the position of a specific workload in both types of queues and list pending workloads in a specific queue.
+
+## Motivation
+
+As presented in [KEP#168](https://github.com/kubernetes-sigs/kueue/tree/main/keps/168-pending-workloads-visibility), there is currently a proposal for a mechanism that supports fetching the order of pending workloads, but it comes with a lot of cons. This proposal addresses all of those problems.
+
+### Cons of the current solution
+
+#### Size of the queue
+
+There are a few scalability concerns. The first one is that the number of fetched pending workloads is limited by the etcd object's size limit. By default, a user is able to fetch only 10 workloads stored at the head of a queue. This number can be increased up to 4000, but comes with a performance loss.
+
+#### Consistency across all LocalQueues
+
+Another scalability drawback is that in a Kueue setup with a lot of LocalQueues it is very likely to hit the Kueue QPS. Assuming Kueue setup with multiple LocalQueues pointing to the same ClusterQueue, Kueue needs to send updates to all LocalQueues to update their status, in order to keep the workload positional information up-to-date. This consumes `QPS` which can lead to blocking other requests. Although we can use a client separate from a default one, it would not not completely resolve all scalability issues
```

Comment:
```suggestion
Another scalability drawback is that in a Kueue setup with a lot of LocalQueues it is very likely to hit the Kueue QPS. Assuming Kueue setup with multiple LocalQueues pointing to the same ClusterQueue, Kueue needs to send updates to all LocalQueues to update their status, in order to keep the workload positional information up-to-date. This consumes `QPS` which can lead to blocking other requests. Although we can use a client separate from a default one, it would not not completely resolve all scalability issues.
```
nit